alchemist/alchemist-fxui/src/main/java/it/unibo/alchemist/boundary/interfaces/FXOutputMonitor.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2010-2020, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.boundary.interfaces;
+
+import it.unibo.alchemist.boundary.gui.effects.EffectGroup;
+import it.unibo.alchemist.boundary.jfx.events.keyboard.KeyboardActionListener;
+import it.unibo.alchemist.model.interfaces.Position2D;
+import javafx.scene.Node;
+
+import java.util.Collection;
+
+/**
+ * {@code OutputMonitor} that handles the graphical part of the simulation in JavaFX.
+ *
+ * @param <T> the {@link it.unibo.alchemist.model.interfaces.Concentration} type
+ * @param <P> the position type
+ */
+public interface FXOutputMonitor<T, P extends Position2D<? extends P>> extends OutputMonitor<T, P> {
+
+    /**
+     * Getter method for the steps.
+     *
+     * @return how many simulation steps this monitor updates the graphics
+     */
+    int getStep();
+
+    /**
+     * Setter method fo the steps.
+     *
+     * @param step How many steps should be computed by the engine for the
+     *             display to update the graphics
+     */
+    void setStep(int step);
```

Comment:
This is only called by AbstractFXDisplay, which implements FXOutputMonitor, in its constructor.

This looks to me like there was an intention to make it so that the monitor could paint the effects every X simulation steps, instead of for every step.
This is not implemented though. Should I remove these leftovers?

Reply:
Yes please